### DISCUSSION TITLE
Deploy Helm Chart on Private EKS Cluster with CDK

### DISCUSSION THREAD

**Discussion by bdudra** [aws/aws-cdk#30727] [Q&A]
Greetings,

I'm attempting to deploy a Helm chart on an EKS cluster with a private endpoint using CDK.  I'm getting a connection timeout when I attempt to deploy the CDK.

I import the cluster using the following CDK:

`const cluster = cdk.aws_eks.FargateCluster.fromClusterAttributes(this, 'cluster', {
      clusterName: `kube-${env}`,
      vpc: vpc,
      securityGroupIds: [eksSgId1, eksSgId2],
      clusterSecurityGroupId: eksSgId1,
      kubectlLambdaRole: cdk.aws_iam.Role.fromRoleArn(this, 'kubectl-lambda-role', kubectlLambdaRoleArn),
      kubectlRoleArn: kubectlRoleArn,
      clusterEndpoint: clusterEndpoint
    });`

And I install using the following CDK:

` const connectedServices = new cdk.aws_eks.HelmChart(this, 'connected-services-helm',{
      cluster: cluster,
      chart: 'connected-services',
      release: 'connected-services',
      namespace: "connected-services",
      createNamespace: false,
      repository: 'private',
      version: '1.2.0',
    });`

But I get the following error message:

`The stack named EksApplicationsStack failed creation, it may need to be manually deleted from the AWS console: ROLLBACK_COMPLETE: Received response status [FAILED] from custom resource. Message returned: Error: b'Error: Kubernetes cluster unreachable: Get "https://private.eks.amazonaws.com/version": dial tcp xxx.xxx.xxx.xxx:443: i/o timeout\n'

Logs: /aws/lambda/EksApplicationsStack-EksApplicatio-Handler886CB40B-pZyH4Nknnjvr

    at invokeUserFunction (/var/task/framework.js:2:6)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async onEvent (/var/task/framework.js:1:369)
    at async Runtime.handler (/var/task/cfn-response.js:1:1676) (RequestId: be6c1a7a-da93-4dbe-8d15-1081fe1fd28c)`

Is there anything that I need to do to enable this process?

**Comment by dhanshew72** [comment]
Any update or workaround for this? Ran into the same issue, haven't found a way to get connection to the cluster out of the box.

**Reply by bdudra** [reply]
I was not able to lookup the EKS cluster and deploy the helm chart. 

My workaround was to simply deploy the Helm chart as part of my CDK stack for the cluster itself or create a dependent stack that references the EKS cluster.  That has worked fine.  It's just made my CDK pretty lengthy. 

Just remember to comment appropriately.

**Reply by dhanshew72** [reply]
I see, it goes through the handler process to update the charts, I ended up using ECR to store my helm charts and passed the URL through. Thank you for the help.
